feat: implementing columns block variant

anuj-adobe opened the pull request:
- Adding following two variants to columns block

 
**1. products**
<img width="1296" alt="image" src="https://github.com/user-attachments/assets/14395629-8a74-4eac-aa8e-8151e4b6cc2b">



**2. benefits**
<img width="1211" alt="image" src="https://github.com/user-attachments/assets/0badc720-8691-4c12-b3f9-00d2e7dc420b">

[Sample Authoring Doc](https://adobe.sharepoint.com/:w:/r/sites/HelixProjects/_layouts/15/Doc.aspx?sourcedoc=%7B9D8C0C52-77BD-4C72-AD5F-66BD38449C6B%7D&file=columns.docx&action=default&mobileredirect=true)

Fix #14 

Test URLs:
- Before: https://main--suncorp--aemsites.hlx.live/_draft/anuj/columns
- After: https://issue-14--suncorp--aemsites.hlx.live/_draft/anuj/columns

JiangLong2019 commented:
Add some comments on look & feel:
**Column products**
1. In mobile view, the title size looks different.
Original site:
<img width="384" alt="image" src="https://github.com/user-attachments/assets/0f710b5c-9862-4a56-a8ba-2eb4b440bf2b">

EDS site:
<img width="308" alt="image" src="https://github.com/user-attachments/assets/e6e08195-bbde-460e-b591-e6b15ae68692">

2. Missing the gray line between columns.
<img width="382" alt="image" src="https://github.com/user-attachments/assets/0165d396-ca5d-4da4-8945-42b843a3fed7">

3. Question - current site, the following buttons are included in the product section. Any idea to handle them?

<img width="382" alt="image" src="https://github.com/user-attachments/assets/0fdf4a7c-d07b-4c2e-8857-fa999711aa30">

4. Comparing with other icons, what's the difference that this one is included directly?
<img width="476" alt="image" src="https://github.com/user-attachments/assets/da03c171-84ac-48d2-9053-84cd80432296">



**Column benefits**
1. Mobile view shows different.
Original site:
<img width="366" alt="image" src="https://github.com/user-attachments/assets/23f87416-325a-40da-a837-0aeeb970f544">

EDS site:
<img width="294" alt="image" src="https://github.com/user-attachments/assets/cc8ada52-5dd7-44e3-a75b-95890694e56c">

2. Tablet view, icon and content are not aligned.
Original site:
<img width="670" alt="image" src="https://github.com/user-attachments/assets/54426d04-0791-4209-a20b-adbca296de99">

EDS site:
<img width="419" alt="image" src="https://github.com/user-attachments/assets/6ae5b090-d50e-42db-bc34-1e44fc3fa419">

JiangLong2019 commented on blocks/columns/columns.css:
```diff
@@ -1,3 +1,27 @@
+  --products-name-height: 5rem; /* 80px */
+  --products-name-font-size-mob-desktop: 1.75rem; /* 28px */
+  --products-name-font-size-tablet: 1.25rem; /* 20px */
+  --products-name-font-color: black;
```
Nit: let's use the global property from styles.css

anuj-adobe replied:
done

JiangLong2019 commented on blocks/columns/columns.js:
```diff
@@ -1,9 +1,127 @@
+
 export default function decorate(block) {
   const cols = [...block.firstElementChild.children];
   block.classList.add(`columns-${cols.length}-cols`);
```
Seems the class is never used.

anuj-adobe replied:
My understanding is to not change the default implementation of columns while implementing new variations.

JiangLong2019 commented on blocks/columns/columns.js:
```diff
@@ -1,9 +1,127 @@
+  if (h3) productNameDiv.appendChild(h3);
+
+  const productDescriptionDiv = document.createElement('div');
+  const descriptionPara = (titleIcon || titlePicture) ? col.querySelector('p:nth-of-type(2)') : col.querySelector('p:first-of-type');
```
Nit: about `p:nth-of-type(2)`, I would use the next element of h3 tag. The assumption is that the icon may not be authored but the title should be there. It depends on how it's authored, maybe not a problem at all.

anuj-adobe replied:
> The assumption is that the icon may not be authored but the title should be there.

I am handling this via conditional code -
1. if icon is present, description will be second <p>
2. if icon is not present, description will be first <p>

Am I missing any other case?

JiangLong2019 replied:
How about this case? -  the icon is not present but the title and description are authored.
*Don't take so much time on this, we can add a description into Sidekick Library to tell customer which field is mandatory.

anuj-adobe commented:
> Question - current site, the following buttons are included in the product section. Any idea to handle them?
image

> 
![image](https://github.com/user-attachments/assets/9869decf-33b5-4966-9f51-cb485dae6f9e)

Probably we can create a new section for these buttons below the products section and add a style to make it visible only in mobile view. wdyt?

anuj-adobe commented:
> 4. Comparing with other icons, what's the difference that this one is included directly?

As discussed in standup, original site is using multiple source of icons -
1. font file 2. raw image 3. svg

I've added icons in EDS as present in original site.
Also, as per our discussion I've initiated a thread with the customer to understand if we can use every icon as svg
here: [Authoring Icons](https://teams.microsoft.com/l/message/19:81uxJmjaWTwuy-ZDVNEb_b-1LHQQBVKm31eRWOZ1px01@thread.tacv2/1724667200339?tenantId=36fb8e1d-a891-493f-96d0-3038e4e9291c&groupId=efba1680-2043-4e5c-8de1-bb7d7f7a78b8&parentMessageId=1724667200339&teamName=aem-suncorp&channelName=General&createdTime=1724667200339)

anuj-adobe commented:
I've addressed the review comments via commit: https://github.com/aemsites/suncorp/pull/35/commits/a484db2ee672893c58284c7574eaccd969a78398 and also replied to few comments.

JiangLong2019 commented:
@anuj-adobe That's a good suggestion, please raise a ticket.

>Probably we can create a new section for these buttons below the products section and add a style to make it visible only in mobile view. wdyt?

JiangLong2019 commented:
Approved. BTW, it's good to convert icon to SVG.

jacobrendina commented on scripts/aem.js:
```diff
@@ -401,12 +401,15 @@ function decorateIcon(span, prefix = '', alt = '') {
-  img.alt = alt;
-  img.loading = 'lazy';
-  span.append(img);
+
```
Given we're no longer going with the icon font approach, should these icon font changes (and the icon font files themselves) be removed from this PR before merging?

jacobrendina commented on scripts/scripts.js:
```diff
@@ -41,6 +41,18 @@ async function loadFonts() {
+/**
+ * load icons.css and set a session storage flag
+ */
+async function loadIcons() {
```
Same story here

anuj-adobe commented:
@JiangLong2019 @jacobrendina 
I have removed the code, earlier added to support multiple icon sources in columns block.
Please have a final look and approve if it seems okay.

jacobrendina commented on styles/styles.css:
```diff
@@ -340,4 +341,18 @@ main .section.highlight {
\ No newline at end of file
+}
+
+main .section.benefits-background-plain {
```
Would be good to merge the latest main branch into your working branch to ensure the tweaks to the section styling are all compatible

anuj-adobe replied:
Latest main is already merged.

jacobrendina reviewed (commented):
Just left one comment suggesting to merge the latest main into this working branch before pushing

JiangLong2019 commented:
@anuj-adobe Some findings.

1. Missing styling, are you planning to add the styling separately from current block?
<img width="693" alt="image" src="https://github.com/user-attachments/assets/c8537764-d5e9-4b0f-a0c0-1a88393745d5">

2. The "Get a Quote" button doesn't work, I think the plan is to wait for Hero block to come in, and then you will merge the modal, right?

anuj-adobe commented:
> @anuj-adobe The "Get a Quote" button doesn't work, I think the plan is to wait for Hero block to come in, and then you will merge the modal, right?

Yes exactly, I am not keeping modal implementation under this PR scope, we can merge it later. I can create an issue if that is not covered under existing list.

anuj-adobe commented:
> @anuj-adobe That's a good suggestion, please raise a ticket.
> 
> > Probably we can create a new section for these buttons below the products section and add a style to make it visible only in mobile view. wdyt?

Created an issue for this -
https://github.com/aemsites/suncorp/issues/48
